linux-packaging-appimage/scripts/deploy.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/sh
+# OpenNox AppImage bulder by Xandros Darkstorm (Discord for support: xandrosdarkstorm)
+# Version 1.2
+
+# Unfortunately AppImageKit project has no strict policy regarding releases.
+# Use "continuous" build to get the latest development build. You can still specify any other build (13 is the latest).
+APPIMAGETOOL_RELEASE="13"
+PROJECT_FOLDER="${GITHUB_WORKSPACE}/build"
+LIBS_LIST="libs_to_copy"
+
+if [ -z $OPENNOX_CFG ]; then
+	OPENNOX_CFG="opennox/opennox.yml"
+fi
+if [ -z $OPENNOX_DATA ]; then
+	OPENNOX_DATA="opennox/gamedata"
+fi
+if [ -z $OPENNOX_STATEFILES ]; then
+	OPENNOX_STATEFILES="opennox"
+fi
+#SIGNKEY="xandros-test-key"
+
+exit_on_error()
+{
+if [ $? -gt 0 ]; then
+echo "$1"
+exit 1
+fi
+}
+
+bundle_struct_exit_on_error()
+{
+exit_on_error "Failed to create bundle structure. Aborting."
+}
+
+bundle_makedir_exit_on_error()
+{
+exit_on_error "Failed to make directory '$1' for library. Aborting."
+}
+
+bundle_filecopy_exit_on_error()
+{
+exit_on_error "Failed to copy files. Aborting."
+}
+
+exit_on_build_fail()
+{
+exit_on_error "Failed to compile AppImage. Aborting."
+}
+
+make_dummyicon_on_fail()
+{
+if [ $? -gt 0 ]; then
+echo "Icon not found. Making a stub icon."
+touch opennox_bundle.AppDir/opennox.png
+fi
+}
+
+echo "Checking appimagetool..."
+if [ ! -x "./appimagetool-$APPIMAGETOOL_RELEASE.AppImage" ] || [ "$APPIMAGETOOL_RELEASE" = "continuous" ]; then
+echo "Downloading appimagetool..."
+wget -q --show-progress  -O appimagetool-$APPIMAGETOOL_RELEASE.AppImage "https://github.com/AppImage/AppImageKit/releases/download/$APPIMAGETOOL_RELEASE/appimagetool-i686.AppImage"
```

Comment:
Maybe we can move it to a `Dockerfile`, so that it's clear that it's a dependency?

Reply:
The reasons why i left this step untouched:
1. Script was not adapted to Docker uniquely. I just changed the place where binaries are kept.
2. While i specified v13 of the appimagetool to be used, it is possible to switch to "continuous" build, which provides latest updates and features. In case of "continuous" build, you have to download the tool every time you use it just to make sure you are using the fresh build.
3. Download times are small, so there is no reason to worry that this packaging step will take a lot of time. We are talking about a matter of seconds here (it was about 3-4 seconds on my machine).

Reply:
I'd prefer to move it to Dockerfile as well - if for some reason the link is invalidated, we would still have time to fix it here instead of having the main opennox build immediately fail.